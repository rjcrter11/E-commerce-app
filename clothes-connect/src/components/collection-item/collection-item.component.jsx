import React from 'react';
import { connect } from 'react-redux';


import CustomButton from '../custom-button/custom-button.component'
import { addItem } from '../../redux/cart/cart.actions'
import './collection-item.styles.scss';


export const CollectionItem = ({ item, addItem }) => {
    const { name, price, imageUrl } = item;
    return (
        <div className='collection-item' >
            <div
                data-testid="background-image"
                className='image'
                style={{
                    backgroundImage: `url(${imageUrl})`
                }}
            />
            <div className='collection-footer' >
                <span data-testid="name" className='name' > {name} </span>
                <span data-testid="price" className='price' > {price} </span>

            </div>
            <CustomButton data-testid='add-button' onClick={() => addItem(item)} inverted >Add to cart</CustomButton>
        </div>
    )
}

const mapDispatchToProps = dispatch => ({
    addItem: item => dispatch(addItem(item))
})

export default connect(
    null,
    mapDispatchToProps
)(CollectionItem)