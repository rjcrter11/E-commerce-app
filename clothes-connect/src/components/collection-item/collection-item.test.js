import React from 'react';

import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import { CollectionItem } from './collection-item.component'


describe('CollectionItem component', () => {
    let wrapper;
    let enzymeWrapper;
    let mockAddItem;
    const imageUrl = 'www.testImg.com';
    const mockName = 'black hat';
    const mockPrice = 10;

    beforeEach(() => {
        mockAddItem = jest.fn()

        const mockProps = {
            item: {
                imageUrl: imageUrl,
                price: mockPrice,
                name: mockName
            },
            addItem: mockAddItem
        };

        wrapper = create(<CollectionItem  {...mockProps} />)
        enzymeWrapper = shallow(<CollectionItem {...mockProps} />)
    });

    it('should render CollectionItem component', () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should call addItem when AddButton is clicked', () => {
        enzymeWrapper.find('[data-testid="add-button"]').simulate('click');
        expect(mockAddItem).toHaveBeenCalled();
    });
    it.skip('should render image as a prop on BackgroundImage', () => {
        expect(enzymeWrapper.find('[data-testid="background-image"]').prop('imageUrl')).toBe(imageUrl)
    })
    it('should render name prop in span', () => {
        expect(enzymeWrapper.find('[data-testid="name"]').text()).toBe(" black hat ")
    })
    it('should render price prop in span', () => {
        const price = parseInt(enzymeWrapper.find('[data-testid="price"]').text());
        expect(price).toBe(mockPrice);
    })
})