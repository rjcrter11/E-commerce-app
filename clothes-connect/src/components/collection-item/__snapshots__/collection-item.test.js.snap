// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem component should render CollectionItem component 1`] = `
<div
  className="collection-item"
>
  <div
    className="image"
    data-testid="background-image"
    style={
      Object {
        "backgroundImage": "url(www.testImg.com)",
      }
    }
  />
  <div
    className="collection-footer"
  >
    <span
      className="name"
      data-testid="name"
    >
       
      black hat
       
    </span>
    <span
      className="price"
      data-testid="price"
    >
       
      10
       
    </span>
  </div>
  <button
    className=" inverted  custom-button"
    data-testid="add-button"
    onClick={[Function]}
  >
    Add to cart
  </button>
</div>
`;
